src/components/sub-menu/SubMenu.vue

```diff
@@ -0,0 +1,96 @@
+<template>
+  <v-menu
+    :close-on-content-click="false"
+    :offset-x="isOffsetX"
+    :offset-y="isOffsetY"
+    :open-on-hover="isOpenOnHover"
+    :transition="transition"
+    :value="openMenu"
+  >
+    <template #activator="{ on }">
+      <v-btn
+        v-if="icon"
+        :color="color"
+        v-on="on"
+      >
+        <v-icon>{{ icon }}</v-icon>
+      </v-btn>
+      <v-list-item
+        v-else-if="isSubMenu"
+        class="d-flex justify-space-between"
+        v-on="on"
+      >
+        {{ name }}
+        <div class="flex-grow-1" />
+        <v-icon>mdi-chevron-right</v-icon>
+      </v-list-item>
+      <v-btn
+        v-else
+        :color="color"
+        text
+        v-on="on"
+        @click="openMenu=true"
+      >
+        {{ name }}
+      </v-btn>
+    </template>
+    <v-list>
+      <template v-for="(item, index) in menuItems">
+        <v-divider
+          v-if="item.isDivider"
+          :key="index"
+        />
+        <sub-menu
+          v-else-if="item.menu"
+          :key="index"
+          :is-open-on-hover="false"
+          :is-offset-x="true"
+          :is-offset-y="false"
+          :is-sub-menu="true"
+          :name="item.name"
+          :menu-items="item.menu"
+          @menu-click="emitClickEvent($event)"
+        />
+        <v-list-item
+          v-else
+          :key="index"
+          @click="emitClickEvent(item)"
+        >
+          <v-list-item-title>{{ item.name }}</v-list-item-title>
+        </v-list-item>
+      </template>
+    </v-list>
+  </v-menu>
+</template>
+
+<script lang="ts">
+import Vue from 'vue'
+import { Component, Prop } from 'vue-property-decorator'
+
+/**
+ * Adapted from https://codepen.io/Moloth/pen/ZEBOzQP.
+ * See also https://github.com/vuetifyjs/vuetify/issues/1877.
+ */
+@Component({})
+export default class SubMenu extends Vue {
+  // props
+  @Prop() readonly name!: string
+  @Prop() readonly icon!: string
+  @Prop() readonly menuItems!: Array<any>
```

Comment:
It's not possible here to specify the type of the Array right Sev? I'm looking and I'm seeing the array is of mixed things.

Reply:
Yes, in this scenario the array is mixed items. One could come up with an interface (possibly a union of interfaces) that would work here, but I don't want to spend any more time on this as I won't be using it.

Reply:
Totally agree. I was just out asking out of curiosity. Alright Sev, resolving.